client/client.go

```diff
@@ -60,31 +71,54 @@ func NewClient(ctx context.Context, apiKey string, orgName string, env string) *
 
 	return &Client{
 		apiKey:      apiKey,
+		orgName:     orgName,
 		baseURL:     baseURL,
-		client:      http.DefaultClient,
+		context:     ctx,
+		rateLimiter: rate.NewLimiter(rate.Limit(DefaultRateLimitPerSecond), 1),
+		client: &retryablehttp.Client{
+			HTTPClient:   http.DefaultClient,
+			CheckRetry:   checkHTTPRetry,
+			RetryWaitMin: 3 * time.Second,
+			Backoff:      retryablehttp.DefaultBackoff,
+			RetryMax:     DefaultRetryMax,
+		},
 		contentType: "application/vnd.api+json",
 	}
 }
 
+// checkHTTPRetry inspects HTTP errors from the Lightstep API for known transient errors
+func checkHTTPRetry(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	if resp.StatusCode == http.StatusInternalServerError {
+		return true, nil
+	}
+	return false, nil
+}
+
 // CallAPI calls the given API and unmarshals the result to into result.
 func (c *Client) CallAPI(httpMethod string, suffix string, data interface{}, result interface{}) error {
 	return callAPI(
-		context.TODO(),
-		c.client,
+		c.context,
+		c,
 		fmt.Sprintf("%v/%v", c.baseURL, suffix),
 		httpMethod,
 		Headers{
-			"Authorization": fmt.Sprintf("bearer %v", c.apiKey),
-			"Content-Type":  c.contentType,
-			"Accept":        c.contentType,
+			"Authorization":   fmt.Sprintf("bearer %v", c.apiKey),
+			"User-Agent":      "terraform-provider-lightstep",
```

Comment:
For future proofing and more analytics, might want to include the provider version too based on the User-Agent [standards](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent)

Reply:
Now with version + terraform sdk version